### PR TITLE
Add test for maintenance.models.Maintenance.run

### PR DESCRIPTION
**Pull request by haniffm**

### CODE REVIEW

File: ESSArch_Core/maintenance/tests/test_models.py [hunk 1]
```diff
@@ -140,6 +141,126 @@ def test_mark_as_complete_when_report_generation_raised_exception(self, mock_gen
...
+        mock__get_report_directory.side_effect = [self.appraisal_path, self.conversion_path]
+
+        self.appraisal_job.run()
+        self.conversion_job.run()
+
+        mock_appr__run.assert_called_once()
+        mock_con__run.assert_called_once()
+        mock__mark_as_complete.assert_called()
+
+        # Update object form DB
```
Comment (OskarPersson) [inline ESSArch_Core/maintenance/tests/test_models.py]:
```suggestion
        # Update object from DB
```

File: ESSArch_Core/maintenance/tests/test_models.py [hunk 2]
```diff
@@ -140,6 +141,126 @@ def test_mark_as_complete_when_report_generation_raised_exception(self, mock_gen
...
+
+        self.reset_access_rights()
+
+        after = timezone.now()
+
+        apr_run.assert_not_called()
+        con_run.assert_not_called()
+        mark_as_complete.assert_not_called()
+
+        # Update object form DB
```
Comment (OskarPersson) [inline ESSArch_Core/maintenance/tests/test_models.py]:
```suggestion
        # Update object from DB
```

File: ESSArch_Core/maintenance/tests/test_models.py [hunk 3]
```diff
@@ -140,6 +141,126 @@ def test_mark_as_complete_when_report_generation_raised_exception(self, mock_gen
...
+        with self.assertRaisesRegex(OSError, f".* No such file or directory: '{self.conversion_path}'"):
+            self.conversion_job.run()
+
+        after = timezone.now()
+
+        apr_run.assert_not_called()
+        con_run.assert_not_called()
+        mark_as_complete.assert_not_called()
+
+        # Update object form DB
```
Comment (OskarPersson) [inline ESSArch_Core/maintenance/tests/test_models.py]:
```suggestion
        # Update object from DB
```

File: ESSArch_Core/util.py [hunk 4]
```diff
@@ -737,3 +738,17 @@ def zip_directory(dirname=None, zipname=None, compress=False):
                 filepath = os.path.join(root, f)
                 arcname = os.path.relpath(filepath, dirname)
                 new_zip.write(filepath, arcname)
+
+
+def has_write_access(directory):
+    if os.name == 'nt':
+        try:
+            tmp_file = os.path.join(directory, str(uuid.uuid4()))
```
Comment (OskarPersson) [inline ESSArch_Core/util.py]: To minimize the risk of the file not being deleted, we should use tempfile.TemporaryFile as a context manager which automatically deletes the file on completion , e.g.

```
with tempfile.TemporaryFile(mode='a', dir=directory) as fp:
    fp.write('')
```

Comment (haniffm) [reply ESSArch_Core/util.py]: Added a comment for the workaround instead. :)

**Pull request merged**